### PR TITLE
124 fix kt table drag and drop reorder behavior

### PR DESCRIPTION
**Pull request by carsoli**

### CODE REVIEW

File: docs/pages/components/tables.vue [hunk 1]
```diff
@@ -404,33 +420,33 @@ You can then use the set sortedColumns prop on the KtTable to update the table u
 </KtTable>
 
 ```html
-<KtTable :rows="rows" :columns="columns">
```
Comment (FlorianWendelborn) [inline docs/pages/components/tables.vue]: Does this example really make sense without `columns`?

File: packages/kotti-banner/src/Banner.vue [hunk 2]
```diff
@@ -65,7 +65,7 @@ export default {
 			}
 		},
 		expandable() {
-			return this.$slots.expand
+			return !!this.$slots.expand
```
Comment (FlorianWendelborn) [inline packages/kotti-banner/src/Banner.vue]:
```suggestion
			return Boolean(this.$slots.expand)
```

File: packages/kotti-table/src/Table.vue [hunk 3]
```diff
@@ -70,10 +70,10 @@ export default {
 
 		loading: Boolean,
 
-		tdClasses: { default: () => [], types: [Array, String, Object] },
```
Comment (FlorianWendelborn) [inline packages/kotti-table/src/Table.vue]: Good catch, I fixed the rest of these on `master` now.

File: packages/kotti-table/src/TableColumn.js [hunk 4]
```diff
@@ -47,6 +47,7 @@ const TableColumn = {
 		renderHeader: Function,
 		renderCell: Function,
 		renderContext: Object,
+		isPropDefined: { default: false, type: Boolean },
```
Comment (FlorianWendelborn) [inline packages/kotti-table/src/TableColumn.js]: random order is best order 😉

File: packages/kotti-banner/src/Banner.vue (hunk 2 above)
Comment (mvangent) [reply packages/kotti-banner/src/Banner.vue]: agreed. @carsoli can you commit this suggestion?

**Comment by mvangent** [comment]
@carsoli code is looking good to me. can you add a test instruction to the description?

File: packages/kotti-banner/src/Banner.vue (hunk 2 above)
Comment (carsoli) [reply packages/kotti-banner/src/Banner.vue]: i don't even know why this commit is here, it should be on #126 :D but i was going to add it to 126...

File: packages/kotti-table/src/TableColumn.js (hunk 4 above)
Comment (carsoli) [reply packages/kotti-table/src/TableColumn.js]: _FEATURE_

File: docs/pages/components/tables.vue (hunk 1 above)
Comment (carsoli) [reply docs/pages/components/tables.vue]: no, u r right

**Comment by carsoli** [comment]
> @carsoli code is looking good to me. can you add a test instruction to the description?
If you follow the issue description, and try to see the expected behavior, that should be it. In the docs it's just `Ordering Columns` subheader that has draggable columns and you should be able to drag as many times as possible without the rendering breaking. (so, in the docs, the formatting of the date should always or the rendering of the avatar as well)

**Comment by mvangent** [comment]
> @carsoli code is looking good to me. can you add a test instruction to the description?
> If you follow the issue description, and try to see the expected behavior, that should be it. In the docs it's just `Ordering Columns` subheader that has draggable columns and you should be able to drag as many times as possible without the rendering breaking. (so, in the docs, the formatting of the date should always work and the rendering of the avatar as well)

hi @carsoli I ran the test. I can verify dragging and dropping is working smoothly. no console.errors. minor: couldn't manage to drag a column to the last row, however from the last row also works.

**Comment by carsoli** [comment]
> > @carsoli code is looking good to me. can you add a test instruction to the description?
> > If you follow the issue description, and try to see the expected behavior, that should be it. In the docs it's just `Ordering Columns` subheader that has draggable columns and you should be able to drag as many times as possible without the rendering breaking. (so, in the docs, the formatting of the date should always work and the rendering of the avatar as well)
> 
> hi @carsoli I ran the test. I can verify dragging and dropping is working smoothly. no console.errors. minor: couldn't manage to drag a column to the last row, however from the last row also works.

I know about the last column thing. it's because of the way the dragging is implemented so in theory it is not a bug, it really is a feature. if you want me to look into allowing for it, i can create a separate feature issue. and yes, the only way to get around this is to drag whatever column to the before-last column THEN drag the last column to the left!

**Comment by mvangent** [comment]
> > > @carsoli code is looking good to me. can you add a test instruction to the description?
> > > If you follow the issue description, and try to see the expected behavior, that should be it. In the docs it's just `Ordering Columns` subheader that has draggable columns and you should be able to drag as many times as possible without the rendering breaking. (so, in the docs, the formatting of the date should always work and the rendering of the avatar as well)
> > 
> > 
> > hi @carsoli I ran the test. I can verify dragging and dropping is working smoothly. no console.errors. minor: couldn't manage to drag a column to the last row, however from the last row also works.
> 
> @mvangent I know about the last column thing. it's because of the way the dragging is implemented so in theory it is not a bug, it really is a feature. if you want me to look into allowing for it, i can create a separate feature issue. and yes, the only way to get around this is to drag whatever column to the before-last column THEN drag the last column to the left!

@carsoli I think its fine to leave it, it was more a heads up, and if you say it is a feature, well... features nowadays..  I approved.

**Pull request merged**